### PR TITLE
Add content on how to collect troubleshooting information

### PR DESCRIPTION
**Pull request by keirthana**
This PR adds additional information about how to collect troubleshooting information from the Anbox deployment and from containers before filing a bug.

### CODE REVIEW

File: howto/troubleshoot/landing.md [hunk 1]
```diff
@@ -1,11 +1,31 @@
 The topics in this guide describe some commonly encountered problems with Anbox Cloud and provide instructions for resolving them. If you encounter an issue with Anbox Cloud, check if any of the following scenarios help in resolving your issue.
 
+[note type="information" status="Note"]If the deployment is older than 3 months, you must upgrade Anbox Cloud to the latest version and see if the required fixes are already part of the upgrade. See [How to upgrade Anbox Cloud](https://discourse.ubuntu.com/t/how-to-upgrade-anbox-cloud/17750) for upgrade instructions.
```
Comment (ajanon) [inline howto/troubleshoot/landing.md:3]: So, just a random thought:

What happens if we change the title of the page 'How to upgrade Anbox Cloud'? Will the link change on the discourse side?
e.g., if we change it to 'How to upgrade', will the link become `https://discourse.ubuntu.com/t/how-to-upgrade/17750`?

If the link does change, doesn't that mean that the previous link will become invalid? Should we remove title pages from link and only keep the thread id? (i.e., `https://discourse.ubuntu.com/t/17750`)

What do you think?

File: howto/troubleshoot/landing.md [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
 * [Troubleshoot issues with initial setup](https://discourse.ubuntu.com/t/troubleshoot-issues-with-initial-setup/35704)
 * [Troubleshoot container failures](https://discourse.ubuntu.com/t/troubleshoot-container-failures/35703)
 * [Troubleshoot issues with application creation](https://discourse.ubuntu.com/t/troubleshoot-issues-with-application-creation/35702)
 * [Troubleshoot issues with LXD clustering](https://discourse.ubuntu.com/t/troubleshoot-issues-with-lxd-clustering/35705)
 
-If you still need help, use the following instructions to collect troubleshooting information from the container and report an issue.
+If you still need help, use any of the following utilities to collect troubleshooting information and report an [issue](https://bugs.launchpad.net/anbox-cloud/+filebug).
+
+
+# Juju crashdump
```
Comment (ajanon) [inline howto/troubleshoot/landing.md:14]:
```suggestion
## Juju crashdump
```

File: howto/troubleshoot/landing.md [hunk 3]
```diff
@@ -1,11 +1,31 @@
...
+* Additional information provided by the Anbox Cloud charms
+* Information about any Anbox containers that crashed
+
+Use the following command to generate a crash dump:
+
+    juju crashdump -s -a debug-layer 
+
+The Anbox Management Service (AMS) charm implements the `debug-layer` addon which will add a `debug-*.tar.gz` archive to the crash dump for the AMS units. The tarball may contain container logs for the containers that are in `error` state in AMS and other information about the Anbox runtime process.
+
+# Anbox Cloud bug report utility
```
Comment (ajanon) [inline howto/troubleshoot/landing.md:28]:
```suggestion
## Anbox Cloud bug report utility
```

File: howto/troubleshoot/landing.md [hunk 4]
```diff
@@ -19,4 +39,12 @@ amc exec <container_id> -- bash -c 'cat "$(anbox-bug-report)"' > "<target_file>"
 ```
 
 This command builds a zip archive that contains the container report. It then
-saves it to the local `<target_file>`. This process might take a few seconds.
\ No newline at end of file
+saves it to the local `<target_file>`. This process might take a few seconds.
+
+# Stored container logs
```
Comment (ajanon) [inline howto/troubleshoot/landing.md:44]:
```suggestion
## Stored container logs
```

File: howto/troubleshoot/landing.md [hunk 5]
```diff
@@ -1,11 +1,31 @@
...
+[note type="information" status="Note"]If the deployment is older than 3 months, you must upgrade Anbox Cloud to the latest version and see if the required fixes are already part of the upgrade. See [How to upgrade Anbox Cloud](https://discourse.ubuntu.com/t/how-to-upgrade-anbox-cloud/17750) for upgrade instructions.
+[/note]
+
 * [Troubleshoot issues with initial setup](https://discourse.ubuntu.com/t/troubleshoot-issues-with-initial-setup/35704)
 * [Troubleshoot container failures](https://discourse.ubuntu.com/t/troubleshoot-container-failures/35703)
 * [Troubleshoot issues with application creation](https://discourse.ubuntu.com/t/troubleshoot-issues-with-application-creation/35702)
 * [Troubleshoot issues with LXD clustering](https://discourse.ubuntu.com/t/troubleshoot-issues-with-lxd-clustering/35705)
 
-If you still need help, use the following instructions to collect troubleshooting information from the container and report an issue.
+If you still need help, use any of the following utilities to collect troubleshooting information and report an [issue](https://bugs.launchpad.net/anbox-cloud/+filebug).
```
Comment (ajanon) [inline howto/troubleshoot/landing.md:11]: So... As a user, what should I do when I see these? Run all three? Run only the relevant ones? How should I know which ones are relevant?

I think we need to provide some guidelines for users. What do you think?

**Review by ajanon — CHANGES_REQUESTED** [review]

File: howto/troubleshoot/landing.md (hunk 1 above)
Comment (keirthana) [reply howto/troubleshoot/landing.md:3]: The link does not change if you change the title. It just takes the title of the post when you create it the first time.

Comment (ajanon) [reply howto/troubleshoot/landing.md:3]: Alright, thanks!

File: howto/troubleshoot/landing.md (hunk 5 above)
Comment (keirthana) [reply howto/troubleshoot/landing.md:11]: I think it depends on what questions support/we ask them. Let me check into this further.

Comment (keirthana) [reply howto/troubleshoot/landing.md:11]: @morphis When you/support have asked for log files or crashdump in the past, what of these three options did you ask? The Juju crashdump (provided the customer has the plugin installed), logs from Anbox report utility, stored container logs. 

We are looking to provide some context on which one to run before filing a bug or will doing all of this help?

Comment (morphis) [reply howto/troubleshoot/landing.md:11]: It depends on where the actual issue is. A crashdump gives you the high level overview, so is definitely a good way to start (or with the appliance `anbox-cloud-appliance.buginfo`).  The other two provide more low level information which are needed in some cases. If someone files a bug the start should always be a crashdump (for regular) or `anbox-cloud-appliance.buginfo` (for appliance): The other two can then be asked for as follow up information.

File: howto/troubleshoot/landing.md [hunk 6]
```diff
@@ -1,11 +1,41 @@
...
+
+
+## Juju crashdump
+
+*Applies to: Anbox Cloud*
```
Comment (morphis) [inline howto/troubleshoot/landing.md:16]: Shouldn't we use something like `Applies to: Juju deployed Anbox Cloud`. "Anbox Cloud" refers to both, the Juju deployed one and the appliance.

**Review by morphis — CHANGES_REQUESTED** [review]

File: howto/troubleshoot/landing.md (hunk 6 above)
Comment (keirthana) [reply howto/troubleshoot/landing.md:16]: No, throughout the docs, we use Anbox Cloud for the regular variant and Anbox Cloud Appliance for the appliance variant. We have the variants explained in line with this in the overview topic (https://anbox-cloud.io/docs/exp/anbox-cloud) as well.

Comment (morphis) [reply howto/troubleshoot/landing.md:16]: Ok, lets go for consistency then. However I have a doubt that people will be able to understand this or read the explanation. Maybe we add a link here to the explanation?

Comment (keirthana) [reply howto/troubleshoot/landing.md:17]: @morphis I have added a note before listing the troubleshooting utilities explaining that there could be variants and also a link to the explanation. Hope this helps.

Comment (morphis) [reply howto/troubleshoot/landing.md:17]: Thanks!

**Pull request merged**